morphia/src/main/java/org/mongodb/morphia/query/QueryImpl.java

```diff
@@ -279,14 +310,17 @@ protected BasicDBObject copy(final BasicDBObject dbObject) {
     }
 
     @Override
+    @Deprecated
     public Query<T> disableCursorTimeout() {
-        noTimeout = true;
+        getOptions().noCursorTimeout(true);
         return this;
     }
 
     @Override
+    @Deprecated
     public Query<T> disableSnapshotMode() {
```

Comment:
Also, BSONObject has a removeField method, so you don't need to get it as a BasicDBObject to remove the $snapshot field.